src/java/org/apache/cassandra/index/sai/QueryContext.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.ReadCommand;
+import org.apache.cassandra.exceptions.QueryCancelledException;
+import org.apache.cassandra.utils.Clock;
+
+/**
+ * Tracks state relevant to the execution of a single query, including metrics and timeout monitoring.
+ *
+ * Fields here are non-volatile, as they are accessed from a single thread.
+ */
+@NotThreadSafe
+public class QueryContext
+{
+    private static final boolean DISABLE_TIMEOUT = Boolean.getBoolean("cassandra.sai.test.disable.timeout");
+
+    private final ReadCommand readCommand;
+    private final long queryStartTimeNanos;
+
+    public final long executionQuotaNano;
+
+    public long partitionsRead = 0;
+    public long rowsFiltered = 0;
+
+    public long queryTimeouts = 0;
+
+    @VisibleForTesting
+    public QueryContext()
```

Comment:
This is only used by `OperationTest#beforeTest`, so unless we have plans to call it more often we could remove this constructor and just add a utility method on `OperationTest`:
```java
private static QueryContext queryContext()
{
    return new QueryContext(null, DatabaseDescriptor.getRangeRpcTimeout(TimeUnit.MILLISECONDS));
}
```
Note also that the null read command might potentially cause a NPE on the call to `QueryCancelledException` at `QueryContext#checkpoint`.

Reply:
That constructor didn't make any sense at all. The usage `OperationTest` was preceded by the creation of a command so I have removed this constructor and am using the main constructor in `OperationTest`. This also removes the possibility of a NPE in the use of `QueryCancelledException`.